06.bowling_object/frame.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require_relative 'shot'
+
+class Frame
+  LAST_FRAME = 9
+  MAX_SHOTS_IN_LAST_FRAME = 3
+  MAX_SHOTS_IN_REGULAR_FRAME = 2
+
+  attr_reader :scores
+
+  def initialize(frame_idx)
+    @scores = []
+    @frame_idx = frame_idx
+  end
+
+  def add_shot(marks)
+    marks.each do |mark|
+      @scores << Shot.new(mark)
+      break if complete?
+    end
+    @scores.size
+  end
+
+  def score(frames)
+    @frame_idx != LAST_FRAME ? @scores.sum(&:score) + bonus_score(frames) : @scores.sum(&:score)
```

Comment:
```suggestion
    @scores.sum(&:score) + bonus_score(frames)
```

最終フレームの扱いはbonus_scoreの中で制御するとここのコードがスッキリしそう